### DISCUSSION TITLE
adding a dataset to EL1000 posthoc

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#244] [Q&A] [answered]
I have an analysis for which EL1000 is nested. I installed it before tsimane2017 became available. When I updated EL1000, the tsimane2017 folder appeared, and it was empty.

When I try to run-procedure setup, I get an error that refers to alphen (the first dataset, which has already been set up):

`$ datalad run-procedure setup`
> [INFO   ] Running procedure setup 
[INFO   ] == Command start (output follows) ===== 
Traceback (most recent call last):
  File "/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/.datalad/procedures/setup.py", line 15, in <module>
    purpose = 'configuration'
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/datalad/distribution/dataset.py", line 583, in require_dataset
    u"{0}.".format(dataset.path))
ValueError: No installed dataset found at /Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/alphen.
[INFO   ] == Command exit (modification check follows) ===== 
CommandError: '/Users/acristia/ChildProjectVenv/bin/python3.6 /Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/.datalad/procedures/setup.py /Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000 ' failed with exitcode 1 under /Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000

I updated my package and did an update -r from my analysis folder; which skips tsimane2017:
> [INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
update(ok): . (dataset)
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
update(ok): EL1000 (dataset)
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/bergelson) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
update(ok): EL1000/bergelson (dataset)
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/elo) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
update(ok): EL1000/elo (dataset)
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/ganek) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
update(ok): EL1000/ganek (dataset)
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/kalashnikova) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
update(ok): EL1000/kalashnikova (dataset)
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/kidd) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
update(ok): EL1000/kidd (dataset)
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/lucid) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
update(ok): EL1000/lucid (dataset)
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/lyon) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
update(ok): EL1000/lyon (dataset)
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/png2019) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
update(ok): EL1000/png2019 (dataset)
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/ramirez-esparza) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/swedish) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/vandam) 
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/warlaumont) 
Fetch:   0%|                                                                                                                                                                                    | 0.00/2.00 [00:00<?, ? Remotes/s]Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Fetch: 100%|Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
[INFO   ] Fetching updates for Dataset(/Users/acristia/Documents/gitrepos/r_avc_cvc/EL1000/winnipeg)                                                                                                                              
Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Warning: untrusted X11 forwarding setup failed: xauth key data not generated

I checked and I do have access to this dataset (I can look at it on the GIN web).

**Comment by lucasgautheron** [comment, marked as answer]
I suggest:

```
git submodule update --init --recursive --remote
```

By the way, you should not need to perform the setup in most cases (unless you need access to x-confidential data)
